storage/conformance_test.go

```diff
@@ -172,7 +174,24 @@ func TestSigningV4Conformance(t *testing.T) {
 					style = BucketBoundHostname(tc.BucketBoundHostname)
 				}
 
-				gotURL, err := SignedURL(tc.Bucket, tc.Object, &SignedURLOptions{
+				if tc.EmulatorHostname != "" {
+					t.Setenv("STORAGE_EMULATOR_HOST", tc.EmulatorHostname)
+				}
+
+				opts := []option.ClientOption{option.WithoutAuthentication()}
+				if tc.ClientEndpoint != "" {
+					opts = append(opts, option.WithEndpoint(tc.ClientEndpoint))
+				}
+				if tc.UniverseDomain != "" {
+					opts = append(opts, option.WithUniverseDomain(tc.UniverseDomain))
+				}
+
+				c, err := NewClient(context.Background(), opts...)
```

Comment:
Is this prone to be impacted by other env options that the user may have set?

Reply:
Hmm... that's a good point. The reason I'm using it here is to mimic a flow with the endpoint, universe domain and emulator env var setting; to make sure that the way it's handled during client creation produces correct hostnames passed through to SignURLs.

I removed the conditional around STORAGE_EMULATOR_HOST which for sure would impact the test - now we are always explicitly setting to the empty string if the test doesn't set an emulator host. There may be others, but none come to mind, and I'm not sure how to test this adequately without creating the client this way.